### PR TITLE
Fix find bug bugs 

### PR DESCRIPTION
**Pull request by DChambersSTFC**
Fixes the following issues from #14

<table border="0" class="bodyTable">
<div class="section">
<h3><a name="org.quattor.ant.NotifyClientTask"></a>org.quattor.ant.NotifyClientTask</h3>
<table border="0" class="bodyTable">
<tr class="a">
<th>Bug</th>
<th>Category</th>
<th>Details</th>
<th>Line</th>
<th>Priority</th></tr>
<tr class="b">
<td>org.quattor.ant.NotifyClientTask.execute() makes inefficient use of keySet iterator instead of entrySet iterator</td>
<td>PERFORMANCE</td>
<td><a class="externalLink" href="http://findbugs.sourceforge.net/bugDescriptions.html#WMI_WRONG_MAP_ITERATOR">WMI_WRONG_MAP_ITERATOR</a></td>
<td>69</td>
<td>Medium</td></tr></table></div><a name="org.quattor.ant.ProfileInfoTask"></a>
<b>Replaced keySet Iteration with EntrySet Iteration, which is more efficient.</b>
<a name="org.quattor.ant.VOConfigTask"></a>
<div class="section">
<h3><a name="org.quattor.ant.VOConfigTask"></a>org.quattor.ant.VOConfigTask</h3>
<table border="0" class="bodyTable">
<tr class="a">
<th>Bug</th>
<th>Category</th>
<th>Details</th>
<th>Line</th>
<th>Priority</th></tr>
<tr class="b">
<td>Unread public/protected field: org.quattor.ant.VOConfigTask.legacySuffixAlgorithm</td>
<td>STYLE</td>
<td><a class="externalLink" href="http://findbugs.sourceforge.net/bugDescriptions.html#URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD">URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD</a></td>
<td>109</td>
<td>Medium</td></tr>
<tr class="a">
<td>Unread public/protected field: org.quattor.ant.VOConfigTask.siteParamsTplNS</td>
<td>STYLE</td>
<td><a class="externalLink" href="http://findbugs.sourceforge.net/bugDescriptions.html#URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD">URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD</a></td>
<td>86</td>
<td>Medium</td></tr></table></div>
<b>Removed unused fields and methods that were made obselete by past commits but were not removed. Corresponding commit for 1 issue: <a href = "https://github.com/quattor/scdb-ant-utils/commit/3e1936cd0277f32517ec558fb45844acb9484d9a">https://github.com/quattor/scdb-ant-utils/commit/3e1936cd0277f32517ec558fb45844acb9484d9a</a></b>
<a name="org.quattor.ant.VOConfigTaskVOCardHandler"></a>
<div class="section">
<h3><a name="org.quattor.ant.VOConfigTaskVOCardHandler"></a>org.quattor.ant.VOConfigTask$VOCardHandler</h3>
<table border="0" class="bodyTable">
<tr class="b">
<th>Bug</th>
<th>Category</th>
<th>Details</th>
<th>Line</th>
<th>Priority</th></tr>
<tr class="a">
<td>Dead store to hostname in org.quattor.ant.VOConfigTask$VOCardHandler.endElement(String, String, String)</td>
<td>STYLE</td>
<td><a class="externalLink" href="http://findbugs.sourceforge.net/bugDescriptions.html#DLS_DEAD_LOCAL_STORE">DLS_DEAD_LOCAL_STORE</a></td>
<td>568</td>
<td>Medium</td></tr></table></div>
<b>Removed assignment of variable that is never used.</b><a name="org.quattor.ant.VOConfigTaskVOConfig"></a>
<div class="section">
<h3><a name="org.quattor.ant.VOConfigTaskVOConfig"></a>org.quattor.ant.VOConfigTask$VOConfig</h3>
<table border="0" class="bodyTable">
<tr class="b">
<th>Bug</th>
<th>Category</th>
<th>Details</th>
<th>Line</th>
<th>Priority</th></tr>
<tr class="a">
<td>org.quattor.ant.VOConfigTask$VOConfig.writeVOTemplate(String) makes inefficient use of keySet iterator instead of entrySet iterator</td>
<td>PERFORMANCE</td>
<td><a class="externalLink" href="http://findbugs.sourceforge.net/bugDescriptions.html#WMI_WRONG_MAP_ITERATOR">WMI_WRONG_MAP_ITERATOR</a></td>
<td>796</td>
<td>Medium</td></tr></table></div><b>Replaced keySet Iteration with EntrySet Iteration, which is more efficient.</b><a name="org.quattor.ant.VOConfigTaskVOMSServer"></a>
<div class="section">
<h3><a name="org.quattor.ant.VOConfigTaskVOMSServer"></a>org.quattor.ant.VOConfigTask$VOMSServer</h3>
<table border="0" class="bodyTable">
<tr class="b">
<th>Bug</th>
<th>Category</th>
<th>Details</th>
<th>Line</th>
<th>Priority</th></tr>
<tr class="a">
<td>Redundant nullcheck of certValue, which is known to be non-null in org.quattor.ant.VOConfigTask$VOMSServer.setOldCert(String)</td>
<td>STYLE</td>
<td><a class="externalLink" href="http://findbugs.sourceforge.net/bugDescriptions.html#RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE">RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE</a></td>
<td>987</td>
<td>Medium</td></tr></table></div><b>Put length check of array as nested within existance check.</b></table>

### CODE REVIEW

File: src/main/java/VOConfigTask.java [hunk 1]
```diff
@@ -551,12 +514,11 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
                             vomsServers.put(VOMSServerKey,vomsServer);
                         }
                         vomsEndpoint.setServer(vomsServers.get(VOMSServerKey));
-                        //vomsEndpoint.setEndpoint("vomss://" + vomsServer.getHost() + ":" + Integer.toString(vomsServer.getPort()) + "/voms/" + voConfig.getName() + "?/" + voConfig.getName());
+                        //vomsEndpoint.setEndpoint("vomss://" + vomsServer.getHost()bae + ":" + Integer.toString(vomsServer.getPort()) + "/voms/" + voConfig.getName() + "?/" + voConfig.getName());
```
Comment (Pansanel) [inline src/main/java/VOConfigTask.java]: Why is the bae string in this comment?

File: src/main/java/ProfileInfoTask.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ public void execute() throws BuildException {
 			throw new BuildException("outputdir (" + outputdir
 					+ ") does not exist");
 		}
-		if (!outputdir.isDirectory()) {
+		if (!outputdir.isDirectory()) {
```
Comment (Pansanel) [inline src/main/java/ProfileInfoTask.java]: Can you remove white spaces after '{' ?

File: src/main/java/VOConfigTask.java (hunk 1 above)
Comment (DChambersSTFC) [inline src/main/java/VOConfigTask.java]: Ah yes I'll remove that. With my current setup, I have issues with the cursor moving without my intent while typing.

**Comment by Pansanel** [comment]
Looks fine for me. Can I merge?

**Comment by Pansanel** [comment]
I simply use the same code as for the other CVM repositories. Pierre will submit a new code to factorize the client.pan code. I will suggest him to use metaconfig.

**Comment by jouvin** [comment]
@Pansanel you probably put your comment in the wrong issue!

**Pull request merged**